PR: Update `RELEASE` notes regarding sdist/wheel building and update installers section (#21911)
spyder-ide/spyder: RELEASE.md

* `git add .` and `git commit -m "Release X.X.X"`

> * python setup.py sdist (needs to be done in a Linux machine to prevent getting permission errors over files. See https://github.com/spyder-ide/spyder/issues/21892 and https://github.com/spyder-ide/spyder/issues/14494)

Reviewer: ```suggestion
* python setup.py sdist

    *Note*: This needs to be done on a Linux machine to prevent getting permission errors on executable files (see https://github.com/spyder-ide/spyder/issues/21892 and https://github.com/spyder-ide/spyder/issues/14494)).
```

Add note on a different line for clarity.

* Activate environment with pip packages only

* pip install -U pip setuptools twine wheel

> * python setup.py bdist_wheel (needs to be done in a Linux machine to prevent getting permission errors over files. See https://github.com/spyder-ide/spyder/issues/21892 and https://github.com/spyder-ide/spyder/issues/14494)

Reviewer: ```suggestion
* python setup.py bdist_wheel

    *Note*: This needs to be done on a Linux machine to prevent getting permission errors in executable files (see https://github.com/spyder-ide/spyder/issues/21892 and https://github.com/spyder-ide/spyder/issues/14494)).
```
